From 5a2d1cc67ba6dbc27754dd6ec56270e221709e3c Mon Sep 17 00:00:00 2001
From: santhosh <santhosh@pinger.com>
Date: Tue, 10 Mar 2015 14:12:45 -0700
Subject: [PATCH] Removed the usage of GetProcessTimes

---
 ffmpeg.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/ffmpeg.c b/ffmpeg.c
index d04f6ab..dda4a1d 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -3821,12 +3821,6 @@ static int64_t getutime(void)
 
     getrusage(RUSAGE_SELF, &rusage);
     return (rusage.ru_utime.tv_sec * 1000000LL) + rusage.ru_utime.tv_usec;
-#elif HAVE_GETPROCESSTIMES
-    HANDLE proc;
-    FILETIME c, e, k, u;
-    proc = GetCurrentProcess();
-    GetProcessTimes(proc, &c, &e, &k, &u);
-    return ((int64_t) u.dwHighDateTime << 32 | u.dwLowDateTime) / 10;
 #else
     return av_gettime_relative();
 #endif
-- 
1.8.4.msysgit.0
